src/import-with-document.ts

```diff
@@ -0,0 +1,5 @@
+import {Document, Import} from 'polymer-analyzer';
+
+export interface ImportWithDocument extends Import {
+  readonly document: Document;
+}
```

Comment:
Can also define a function that will tell you if you have an ImportWithDocument:

```
function importHasDocument(import_: Import): import_ is ImportWithDocument {
  return document !== undefined;
}
```

Then we can use this as a type safe alternative to casting with `as` (which should be avoided as much as possible).

Reply:
Very cool; I was trying out Flow recently and was trying to figure out how to write a type checking function but couldn't get past the fact that the narrowing only applied within the function. :/ It's neat that TypeScript has an explicit way to do this. I've updated the PR with an `isImportWithDocument` function.

Reply:
That flow doc is neat!

I didn't realize that they did type checking of function parameters by looking at everywhere it's called. That seems like too much inference to me, but that might just be change aversion talking.

I really wish that `Array#filter` would flow through predicate functions in typescript.

Reply:
> I really wish that `Array#filter` would flow through predicate functions in typescript.

Same; this is actually why I changed `DocumentConverter#getHtmlImports` to a for loop in this PR.